bin/desi_compute_pixel_flatfield

```diff
@@ -1,95 +1,55 @@
+import scipy.ndimage as ndi
+from numba import cfunc, carray
+from numba.types import intc, intp, float64, voidptr
+from numba.types import CPointer
+from scipy import LowLevelCallable
+
+@cfunc(intc(CPointer(float64), intp,
+            CPointer(float64), voidptr))
+def maskedmed(values_ptr, len_values, result, data):
+    values = carray(values_ptr, (len_values,), dtype=float64)
+    ii=(values!=0)
+    if np.sum(ii)>0 :
+        result[0] = np.median(values[ii])
+    else :
+        result[0] = 0.
+    return 1
+
+def maskedmedian(image,shape) :
+    """ Returns a masked sliding median filtering of the input image. 
+        Zero values in the input image are ignored in the median.
 
     Args:
-        filename : a fits file withe a specter PSF
+        image : 2D np.array
+        shape : tuple of length 2 giving the shape of the filtering window.
 
     Returns:
-        x     : 1D array, x_ccd coordinate of spots (fiber axis, axis=1 in np.array)
-        y     : 1D array, y_ccd coordinate of spots (wavelength axis, axis=0 in np.array)
-        dx/dy : 1D array, dx/dy for curves of constant fiber coordinate
-        dy/dx : 1D array, dy/dx for curves of constant wavelength
+        2D np.array of same shape as input image
     """
-    try :
-        psftype=pyfits.open(filename)[0].header["PSFTYPE"]
-    except KeyError :
-        psftype=""
-
-    psf=None
-
-    if psftype=="GAUSS-HERMITE" :
-        psf=specter.psf.GaussHermitePSF(filename)
-    elif psftype=="SPOTGRID" :
-        psf=specter.psf.SpotGridPSF(filename)
-
-    if psf is None :
-        raise ValueError("cannot read PSFTYPE=%s"%psftype)
-
-    # make a grid of points from this PSF
-    x=[]
-    y=[]
-    f=[]
-    w=[]
-    wstep=50.
-    waves=np.linspace(psf.wmin,psf.wmax,int((psf.wmax-psf.wmin)/wstep))
-    fibers=np.arange(psf.nspec)
-    for fiber in fibers :
-        for wave in waves :
-            tx,ty = psf.xy(fiber,wave)
-            x.append(tx)
-            y.append(ty)
-            f.append(fiber)
-            w.append(wave)
-
-    x=np.array(x)
-    y=np.array(y)
-    dxdy=np.zeros(x.size).astype(float)
-    dydx=np.zeros(x.size).astype(float)
-
-    # use this grid of points to determine dx/dy along wave and dy/dx along fiber
-    for fiber in fibers :
-        mask=np.where(f==fiber)[0]
-        tx=x[mask]
-        ty=y[mask]
-        i=np.argsort(ty)
-        dx=np.gradient(tx[i])
-        dy=np.gradient(ty[i])
-        dxdy[mask[i]]=dx/dy
-
-    for wave in waves :
-        mask=np.where(w==wave)[0]
-        tx=x[mask]
-        ty=y[mask]
-        i=np.argsort(tx)
-        dx=np.gradient(tx[i])
-        dy=np.gradient(ty[i])
-        dydx[mask[i]]=dy/dx
-    return x,y,dxdy,dydx
-
-def median_image(image_filenames) :
-    """ Return a median of input images after rescaling each image
+    return ndi.generic_filter(image, LowLevelCallable(maskedmed.ctypes), size=shape)
+
+
+
+
+def clipped_mean_image(image_filenames) :
```

Comment:
I find it odd we are re-creating a basic task in `astropy`:

https://docs.astropy.org/en/stable/api/astropy.stats.sigma_clip.html

Odds are theirs is better tested and debugged..